### PR TITLE
[PROD-4862]: Add beneficiaryRoles to BeneficiariesCredential schema and update related files

### PR DESCRIPTION
**Pull request by ShivamKumarSa**
# Description

- Introduced `beneficiaryRoles` field in the BeneficiariesCredential schema, allowing for a comma-separated string format.
- Updated the beneficiaries.md documentation to include details about `beneficiaryRoles`.
- Modified the BeneficiariesCredential JSON and JSON-LD schemas to reflect the new field.
- Enhanced test scenarios to validate the new `beneficiaryRoles` field, including various edge cases and expected failures for invalid inputs.

Fixes # [PROD-4862](https://redbelly.atlassian.net/browse/PROD-4862)
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Test
- [ ] Integration Test
- [ ] Manual Test
- [ ] Others (specify)

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

[PROD-4862]: https://redbelly.atlassian.net/browse/PROD-4862?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: credentials/beneficiaries.md [hunk 1]
```diff
@@ -5,3 +5,13 @@ The names of Beneficiaries linked to the business in a comma separated format.
...
 ```
+
+# beneficiaryRoles
+
+The roles of Beneficiaries linked to the business in a &(ampersand) separated format and each beneficiary roles seperated by commma.
+
+Like if there are 2 users and 1st one has ubo and director roles and 2nd one has ubo and representative roles
+
+```
+  "beneficiaryRoles": "ubo&director,ubo&representative"
```
Comment (norman-atashbar-rb) [inline credentials/beneficiaries.md:16]: It is possible that a role has & in it (like R&D lead, manager&director), if it is not a closed list. If this is the case, I would pick a different delimiter (like `|`).

Comment (norman-atashbar-rb) [reply credentials/beneficiaries.md:16]: Actually, I see the list is closed, so all good here.

File: schemas/json/BeneficiariesCredential.json [hunk 2]
```diff
@@ -22,33 +22,21 @@
   ],
   "properties": {
     "@context": {
-      "type": [
-        "string",
-        "array",
-        "object"
-      ]
+      "type": ["string", "array", "object"]
```
Comment (norman-atashbar-rb) [inline schemas/json/BeneficiariesCredential.json:25]: We should add a linter rule that applies consistent formatting. We don't want another PR that reverts this formatting again.

File: test/tools/proofOfIncorporationTestScenarios.ts [hunk 3]
```diff
@@ -57,7 +57,9 @@ function generateProofOfIncorporationCredential(
     id: did,
     legalName: faker.company.name(),
     legalAddress: `${faker.location.streetAddress()}, ${faker.location.city()}, ${faker.location.country()}`,
-    registrationNumber: faker.string.alphanumeric({ length: { min: 6, max: 12 } }),
```
Comment (norman-atashbar-rb) [inline test/tools/proofOfIncorporationTestScenarios.ts:60]: Lack of linting rule for TS code as well, unnecessarily blowing the size of the PR.

File: test/tools/proofOfIncorporationTestScenarios.ts [hunk 4]
```diff
@@ -267,14 +273,15 @@ export const proofOfIncorporationTestScenarios = [
...
-      data.credentialSubject.legalForm = 'Etaireia Periorismenis Efthynis (E.P.E.) Ltd';
+      data.credentialSubject.legalForm =
+        'Etaireia Periorismenis Efthynis (E.P.E.) Ltd';
     }),
     expectedValid: true,
   },
   {
     name: 'Valid legalForm with ampersand',
     data: generateProofOfIncorporationCredential((data) => {
-      data.credentialSubject.legalForm = 'Smith & Associates, LLC';
```
Comment (norman-atashbar-rb) [inline test/tools/proofOfIncorporationTestScenarios.ts:277]: An interesting use of both , and & characters
Is comma a valid character for the names? If so, perhaps we shouldn't use it to separate values.

File: credentials/beneficiaries.md (hunk 1 above)
Comment (ShivamKumarSa) [reply credentials/beneficiaries.md:16]: https://docs.sumsub.com/reference/get-applicant-data#beneficiaries-element-fields

types field not have & in it they are single word values

File: schemas/json-ld/BeneficiariesCredential.jsonld [hunk 5]
```diff
@@ -18,9 +18,14 @@
             "@id": "beneficiaries-vocab:beneficiaryNames",
             "@type": "xsd:string",
             "@container": "@set"
+          },
```
Comment (norman-atashbar-rb) [inline schemas/json-ld/BeneficiariesCredential.jsonld:21]: Do we need to version these schemas? Can we version them?
These schemas can be used to define the contract for external VC issuance.

Comment (ShivamKumarSa) [reply schemas/json-ld/BeneficiariesCredential.jsonld:21]: yep they already have versions
"@version": 1.1,
we are using 1.1 for all

File: test/tools/proofOfIncorporationTestScenarios.ts (hunk 4 above)
Comment (ShivamKumarSa) [reply test/tools/proofOfIncorporationTestScenarios.ts:277]: these are the test cases for some other credential, it was not added in test cases file so added that and its a different test case of some other vc from our ticket and it supports ampersand

File: schemas/json/BeneficiariesCredential.json (hunk 2 above)
Comment (ShivamKumarSa) [reply schemas/json/BeneficiariesCredential.json:25]: Added format:check in workflow as well for not updating formatting again

File: test/tools/proofOfIncorporationTestScenarios.ts (hunk 3 above)
Comment (ShivamKumarSa) [reply test/tools/proofOfIncorporationTestScenarios.ts:60]: Added format:check in workflow as well for not updating formatting again

**Review by norman-atashbar-rb — APPROVED** [review]
